nixos/modules/config/system-libs.nix

```diff
@@ -0,0 +1,114 @@
+# This module defines the packages that appear in
+# /run/current-system/sw.
+
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.libraries;
+
+  pathsToLink = [ "/lib" "/share" ];
+  mkLibPath = pkgs: pkgs.buildEnv {
+    name = "libraries";
+    paths = cfg.unsafePackages pkgs;
+    inherit pathsToLink;
+    ignoreCollisions = true;
+  };
+  mkSanitized = pkgs: pkgs.buildEnv {
+    name = "sanitized-libraries";
+    paths = cfg.packages pkgs;
+    inherit pathsToLink;
+    ignoreCollisions = true;
+    postBuild = ''
+      for i in $out/lib/*; do
+        if [ -f "$(readlink -f "$i")" ]; then
+          rm "$i"
+        fi
+      done
+    '';
+  };
+
+  libPath = mkLibPath pkgs;
+  libPath32 = mkLibPath pkgs.pkgsi686Linux;
+
+  # We use more "stable" absolute paths in systemd environment and symlinks for users.
+  mkEnv = path: path32: mapAttrs (name: concatMap (suffix: [ "${path}/${suffix}" ] ++ optional cfg.support32Bit "${path32}/${suffix}")) cfg.environment;
+
+in
+
+{
+  options = {
+
+    libraries = {
+
+      packages = mkOption {
+        type = types.function types.attrs (types.listOf types.package);
+        default = pkgs: [];
```

Comment:
The idea is that this function returns needed packages from passed package set. So when we need 32-bit support we evaluate this function twice, first with `pkgs` and then with `pkgsi686Linux`, getting what we want.